lale/lib/lale/aggregate.py

```diff
@@ -12,16 +12,122 @@
+    from pyspark.sql.dataframe import DataFrame as spark_df
+
+    spark_installed = True
+
+except ImportError:
+    spark_installed = False
+
+
+def _is_pandas_df(df):
+    return isinstance(df, pd.core.groupby.generic.DataFrameGroupBy) or isinstance(
+        df, pd.DataFrame
+    )
+
+
+def _is_spark_df(df):
+    if spark_installed:
+        return isinstance(df, pysql.GroupedData) or isinstance(df, spark_df)  # type: ignore
+
+
+def _is_ast_subscript(expr):
+    return isinstance(expr, ast.Subscript)
+
+
+def _is_ast_attribute(expr):
+    return isinstance(expr, ast.Attribute)
+
 
 class _AggregateImpl:
-    def __init__(self, columns, group_by=[]):
-        self._hyperparams = {"columns": columns, "group_by": group_by}
+    def __init__(self, columns=None, group_by=None):
+        self.columns = columns
+        self.group_by = group_by
+
+    # Commented the validation for now to pass the OBM test cases.
+    # We can uncomment this when OBM starts supporting the new format of Aggregate operator.
+    # @classmethod
+    # def validate_hyperparams(cls, group_by=None, **hyperparams):
+    #     if group_by is not None:
+    #         raise ValueError(
+    #             "The use of group_by in Aggregate is deprecated. Please use the GroupBy operator instead."
+    #         )
 
     def transform(self, X):
-        raise NotImplementedError()
+        agg_info = {}
+        agg_expr = {}
+
+        def create_spark_agg_expr(new_col_name, agg_col_func):
+            functions_module = importlib.import_module("lale.lib.lale.functions")
+
+            def get_spark_agg_method(agg_method_name):
+                if agg_method_name == "sum":
+                    return getattr(functions_module, "grouped_sum")
+                if agg_method_name == "max":
+                    return getattr(functions_module, "grouped_max")
+                if agg_method_name == "min":
+                    return getattr(functions_module, "grouped_min")
+                if agg_method_name == "count":
+                    return getattr(functions_module, "grouped_count")
+                if agg_method_name == "mean":
+                    return getattr(functions_module, "grouped_mean")
+
+            agg_method = get_spark_agg_method(agg_col_func[1])()  # type: ignore
+            return agg_method(agg_col_func[0]).alias(new_col_name)
+
+        for new_col_name, expr in (
+            self.columns.items() if self.columns is not None else []
+        ):
+            agg_func = expr._expr.func.id
+            expr_to_parse = expr._expr.args[0]
+            if _is_ast_subscript(expr_to_parse):
+                agg_col = expr_to_parse.slice.value.s  # type: ignore
+            elif _is_ast_attribute(expr_to_parse):
+                agg_col = expr_to_parse.attr
+            else:
+                raise ValueError(
+                    "Aggregate 'columns' parameter only supports subscript or dot notation for the key columns. For example, it.col_name or it['col_name']."
```

Comment:
Codecov is complaining that the error condition is not covered by tests, so may be just add a test case that captures the error case?